geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/IngestCommand.scala

```diff
@@ -238,47 +238,117 @@ object IngestCommand extends LazyLogging {
 
     var converter: Config = Option(params.config).map(CLArgResolver.getConfig).orNull
 
+    // if there is no converter passed in
     if (converter == null && inputs.nonEmpty) {
-      // if there is no converter passed in, try to infer the schema from the input files themselves
-      Command.user.info("No converter defined - will attempt to detect schema from input files")
-      val file = inputs.iterator.flatMap(PathUtils.interpretPath).headOption.getOrElse {
-        throw new ParameterException("Parameter <files> did not evaluate to anything that could be read")
-      }
+      var open: () => InputStream = null
+      var file: FileHandle = null
       val opened = ListBuffer.empty[CloseableIterator[InputStream]]
-      def open(): InputStream = {
-        val streams = file.open.map(_._2)
-        opened += streams
-        if (streams.hasNext) { streams.next } else {
+
+      // try to infer the schema from stdin
+      if (inputs == Inputs.StdInInputs) {
+        Command.user.info("No converter defined - will attempt to detect schema from stdin")
+
+        // Pass over the first line in stdin - it represents the column header and not a SimpleFeature
+        scala.io.StdIn.readLine()
+
+        /*
+        The act of inferring the schema consumes SimpleFeatures from stdin, meaning that those SimpleFeatures would
+        no longer be in stdin for the ingest to read. To prevent this, we create two identical iterators. One iterator
+        will process the SimpleFeatures to be used for inference, and the other iterator will be used to ingest those
+        SimpleFeatures.
+         */
+        val stdinIterators = Iterator.continually(scala.io.StdIn.readLine().toByte).duplicate
+
+        // For schema inference, we don't need all the SimpleFeatures - just use the first *geomesa.convert.infer.sample* records as a sample
+        val (allFeatures, sampledFeatures) = stdinIterators match {
+          case (a, s) => (a, s.take(AbstractConverterFactory.inferSampleSize))
+        }
+
+        // Function that returns an input stream containing the data from stdin
+        def openStdin(): InputStream = {
+          try {
+            // Create an input stream of the sampled SimpleFeatures
+            val inputStream = new ByteArrayInputStream(sampledFeatures.toStream.toArray)
+            val inputStreamIter = CloseableIterator.single(inputStream)
+            opened += inputStreamIter
+
+            // Return this stream
+            opened.flatten.head
+          } catch {
+            case e: Exception => throw new RuntimeException("Unable to read from stdin", e)
+          }
+        }
+
+        open = openStdin
+      }
```

Comment:
> `AbstractConverterFactory.inferSampleSize` provides a configuration for how many lines should be considered in type inference.
> 
> I would suggest something like defining an input iterator near where your `var open` is. You can duplicate the input stream and sample one duplicate for use in the inferencer and save one for ingest later.
> 
> ```
> val (input, sample) = Iterator.continually(io.StdIn.readLine()).duplicate match { 
>   case (i, s) => (i, s.take(AbstractConverterFactory.inferSampleSize).toList)
> }
> ```

@aheyne I made a change based on your suggestion, but somehow it's unable to parse the SimpleFeature:

```
java.lang.RuntimeException: Unable to read from stdin

	at org.locationtech.geomesa.tools.ingest.IngestCommand$.openStdin$1(IngestCommand.scala:278)
	at org.locationtech.geomesa.tools.ingest.IngestCommand$.$anonfun$getSftAndConverter$5(IngestCommand.scala:282)
	at org.locationtech.geomesa.convert2.SimpleFeatureConverter$.$anonfun$infer$2(SimpleFeatureConverter.scala:136)
	at org.locationtech.geomesa.convert2.SimpleFeatureConverter$.$anonfun$infer$1(SimpleFeatureConverter.scala:136)
	at org.locationtech.geomesa.convert2.SimpleFeatureConverter$.infer(SimpleFeatureConverter.scala:135)
	at org.locationtech.geomesa.tools.ingest.IngestCommand$.$anonfun$getSftAndConverter$1(IngestCommand.scala:307)
	at org.locationtech.geomesa.tools.ingest.IngestCommand$.getSftAndConverter(IngestCommand.scala:233)
	at org.locationtech.geomesa.tools.ingest.IngestCommand.$anonfun$execute$6(IngestCommand.scala:107)
	at org.locationtech.geomesa.tools.ingest.IngestCommand.$anonfun$execute$6$adapted(IngestCommand.scala:105)
	at org.locationtech.geomesa.tools.package$DataStoreCommand.withDataStore(package.scala:68)
	at org.locationtech.geomesa.tools.package$DataStoreCommand.withDataStore$(package.scala:66)
	at org.locationtech.geomesa.accumulo.tools.ingest.AccumuloIngestCommand.withDataStore(AccumuloIngestCommand.scala:20)
	at org.locationtech.geomesa.tools.ingest.IngestCommand.execute(IngestCommand.scala:105)
	at org.locationtech.geomesa.tools.ingest.IngestCommand.execute$(IngestCommand.scala:58)
	at org.locationtech.geomesa.accumulo.tools.ingest.AccumuloIngestCommand.execute(AccumuloIngestCommand.scala:25)
	at org.locationtech.geomesa.accumulo.tools.ingest.IngestCommandTest.$anonfun$new$20(IngestCommandTest.scala:123)
Caused by: java.lang.NumberFormatException: For input string: "26236,Hermione,25,2015-06-07,"Edward, Bill, Harry",40.232,-53.2356"
	at org.locationtech.geomesa.tools.ingest.IngestCommand$.$anonfun$getSftAndConverter$3(IngestCommand.scala:260)
	at org.locationtech.geomesa.tools.ingest.IngestCommand$.openStdin$1(IngestCommand.scala:271)
	... 15 more
```

Reply:
I think it's probably because you're calling .toByte on 260